pkg/scalers/datadog_scaler.go

```diff
@@ -60,17 +61,53 @@ func NewDatadogScaler(ctx context.Context, config *ScalerConfig) (Scaler, error)
 	}, nil
 }
 
+// parseAndTransformDatadogQuery checks correctness of the user query and adds rollup if not available
+func parseAndTransformDatadogQuery(q string, age int) (string, error) {
+	aggregator := regexp.MustCompile(`^(avg|sum|min|max):.*`)
```

Comment:
Maybe, this regex could be moved at package level to avoid compilation on every call.
You can find an example [here](https://github.com/kedacore/keda/blob/main/pkg/scalers/rabbitmq_scaler.go#L25-L29)

WDYT?

Reply:
👍